File: codeforces/vova_game/vova_game.js
var q = +readline();

for(var i = 0; i < q; i++) {
    var arr = readline().split(' ').map(x => +x);
    var k = arr[0];
    var n = arr[1]
    var a = arr[2];
    var b = arr[3];

    if (b * n >= k) {
        write(-1, '\n');
    } else if (a * n < k) {
        write(n, '\n');
    } else {
        var x = (k - b*n) / (a - b);

        write(Math.floor(x), '\n');
    }
}

File: codeforces/vova_game/vova_game.py
import math

q = int(input())

for i in range(q):
    k, n, a, b = [int(x) for x in input().split()]
    
    if b*n >= k:
        print(-1)
    elif a*n < k:
        print(n)
    else:
        x = (k - b*n) / (a - b);

        print(math.ceil(x) - 1)